salt/_modules/metalk8s_drain.py

```diff
@@ -313,8 +244,10 @@ def get_pod_controller(self, pod):
         controller_ref = _get_controller_of(pod)
         if controller_ref is None:
             return None
-        response = self.get_controller(pod.metadata.namespace, controller_ref)
-        if self.not_found(response):
+        response = self.get_controller(
+            pod['metadata']['namespace'], controller_ref
+        )
+        if not response:
```

Comment:
`self.get_controller` is just a `metalk8s_kubernetes.get_object`, which will return `None` or an object. This test (and others) should be instead `if response is not None:`, IMO.

Reply:
If for what ever reason one day `python-kubernetes` return an empty dict instead of `None` we will fail here.
IMO we have no reason to only check for `None` if we have no object then we raise `Missing pod blablabla`

Reply:
We're calling into our own custom module, so we control the fact that `None` is an expected answer for "not found" objects. If instead we want to raise a `NotFound` error, fine, but then there shouldn't be a test for `None` or falsy values either.

Reply:
But why we should only check for `None` ?
If we get `0` or `''` or `[]` or `{}` here we still have an issue even if it should not happen it's maybe better to fail here than "let see what will happen"

Reply:
Because we should not get a falsy value out of our custom module. If we do, it's an internal bug, and we want to catch it.